library/src/scala/quoted/QuoteContext.scala

```diff
@@ -46,8 +51,18 @@ class QuoteContext(val tasty: scala.tasty.Reflection) {
     tasty.warning(msg, expr.unseal.pos) given rootContext
   }
 
+  /** Unpickle `pickledExpr` which represents a pickled `Expr` tree
+   */
+  private[scala] def unpickleExpr[T](pickledExpr: PickledExpr, args: PickledExprArgs): given QuoteContext => Expr[T]
+
+  /** Unpickle `pickledType` which represents a pickled `Type` tree
+   */
+  private[scala] def unpickleType[T](pickledType: PickledType, args: PickledTypeArgs): given QuoteContext => Type[T]
+
```

Comment:
If they are private, why we need them in the trait?

Reply:
The alternative is to move it to the Kernel. That would probably simplify it a bit.